postreise/plot/plot_curtailment_ts.py

```diff
@@ -0,0 +1,201 @@
+import matplotlib.pyplot as plt
+import pandas as pd
+from powersimdata.network.usa_tamu.constants.plants import (
+    type2color,
+    type2label,
+)
+
+from postreise.analyze.check import (
+    _check_resources_and_format,
+    _check_scenario_is_in_analyze_state,
+)
+from postreise.analyze.demand import get_demand_time_series
+from postreise.analyze.generation.curtailment import (
+    get_curtailment_time_series,
+)
+from postreise.analyze.generation.summarize import (
+    get_generation_time_series_by_resources,
+)
+from postreise.analyze.time import (
+    change_time_zone,
+    resample_time_series,
+    slice_time_series,
+)
+
+
+def plot_curtailment_time_series(
+    scenario,
+    area,
+    resources,
+    area_type=None,
+    time_range=None,
+    time_zone="utc",
+    time_freq="H",
+    show_demand=True,
+    percentage=True,
+    t2c=None,
+    t2l=None,
+    title=None,
+    label_fontsize=20,
+    title_fontsize=22,
+    tick_fontsize=15,
+    legend_fontsize=18,
+    save=False,
+    filename=None,
+    filepath=None,
+):
+    """Generate time series curtailment plot of each specified resource in a certain
+    area of a scenario.
+
+    :param powersimdata.scenario.scenario.Scenario scenario: scenario instance
+    :param str area: one of: *loadzone*, *state*, *state abbreviation*,
+        *interconnect*, *'all'*
+    :param str/list resources: one or a list of resources.
+    :param str area_type: one of: *'loadzone'*, *'state'*,
+        *'state_abbr'*, *'interconnect'*
+    :param tuple time_range: (start time stamp, end time stamp). If None, the entire
+        time range is used for the given scenario.
```

Comment:
I think it would be helpful here to list the allowable type(s) of the timestamps.

Reply:
What is the classic doc string format when specifying types of parameters in a tuple, maybe something like this: `:param (pandas.Timestamp/numpy.datetime64/datetime.datetime, pandas.Timestamp/numpy.datetime64/datetime.datetime) time_range:`? @rouille

Reply:
Or `:param tuple time_range: [blah blah blah] where each time stamp is pandas.Timestamp/numpy.datetime64/datetime.datetime`. Does this also work with a str like `"2016-01-01 12:00:00"`?

Reply:
Yeah, keep `:param tuple time_range:` and mention the type of the elements in the description as @danielolsen said.

Personally I would not support a `str` ("2016-01-01 12:00:00"), it is a recipe for failure and it is very easy to use `pd.Timestamp`, which by the way accept "2016-01-01 12:00:00":
```
>>> import pandas as pd
>>> pd.Timestamp("2016-01-01 12:00:00")
Timestamp('2016-01-01 12:00:00')
```

Reply:
Done.